### PR TITLE
54 new folder structure

### LINKED ISSUE #54: Refactor folder structure and apply react file name conventions
As a Team, we'd like to define a shared folder structure and apply it to the existing code.

---

## :white_check_mark:  &nbsp; Conditions of satisfaction

- [ ] there should be a concept (MD) about the long-term folder structure,  
which is discussed & shared with the team
- [ ] Should have a refactored folder structure
- [ ] Should refactor all files to react naming conventions
- [ ] All folders should have speaking identifiers
- [ ] Should have refactored existing files to fit the new folder structure

### PR DESCRIPTION
**Pull request by WolfgangDaniel**
Fixes #54 

## Changes
The project wasn't structured properly preventing us to write scalable and flexible code.
Now the folder structure was completely renewed and the files were partly refactored to fit the new folder structure and the naming conventions. 

## Checklist before merge

**Developer's responsibilities**
* [ ] Approved by at least one developer: *enter reviewer here*
* [ ] **Pull request build** has passed
* [ ] Production backend has been updated to reflect new features and fix bugs
* [ ] tested locally (in all browsers if frontend)
* [ ] updated the docs
* [ ] added tests where necessary

**Product Owner's responsibilities**
* [ ] No issues have been found on staging environment
* [ ] Product Owner has approved the release

### CODE REVIEW

File: client/src/components/content/PropertiesPanel/PropertiesPanelBuilder/PropertiesPanelBuilder.jsx [hunk 1]
```diff
@@ -87,27 +95,30 @@ export default class PropertiesPanelBuilder extends Component {
...
+      sessionStorage.getItem('TaskToApplicationCache')
+    );
 
     if (selectedApplication in currentSavedTasksObject) {
       this.setState({
         selectedApplication: selectedApplication,
-        tasksForSelectedApplication: currentSavedTasksObject[selectedApplication],
-        disableTaskSelection: false
+        tasksForSelectedApplication:
+          currentSavedTasksObject[selectedApplication],
```
Comment (kej-jay) [inline client/src/components/content/PropertiesPanel/PropertiesPanelBuilder/PropertiesPanelBuilder.jsx:120]: I don't see how this linebreak increases readability

File: server/package.json [hunk 2]
```diff
@@ -2,7 +2,7 @@
   "name": "ark-automate",
   "version": "1.0.0",
   "scripts": {
-    "start": "node server",
+    "start": "node server.js",
```
Comment (kej-jay) [inline server/package.json:5]: I am currently not able to test if this works. Why is this change needed?

File: server/routes/rpaFramework.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  '/commands/get-available-applications',
+  commandsController.getAvailableApplications
+);
+
+router.get(
+  '/commands/get-available-tasks-for-application',
+  commandsController.getAvailableTasksForApplications
+);
+
+router.get('/commands/get-vars-for-task', commandsController.getVarsForTask);
```
Comment (kej-jay) [inline server/routes/rpaFramework.js:19]: Shouldn't styling be constistent?

File: server/server.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  const app = express();
+
+  // Priority serve any static files.
+  app.use(express.static(path.resolve(__dirname, '../client/build')));
+  app.use(express.json());
+
+  // Main requests
+  app.use('/rpa-framework', rpaFrameworkRouter);
+
+  // All remaining requests return the React app, so it can handle routing.
```
Comment (kej-jay) [inline server/server.js:33]: What is this comment about?

**Review by kej-jay — CHANGES_REQUESTED** [review]
I am not a big fan of those dummy files. Please check, if there is any way to not have them in the repo, as they are not useful and might confuse people.

As my understanding of the client/api/ folder is that it shall include all API related functions, why are those calls from the front- to the backend not in there, but instead is a dummy file? Please let me know if I understood the purpose of that folder wrong or not.

Other than that only minor things that came up for me

File: server/routes/rpaFramework.js (hunk 3 above)
Comment (lukashueller) [reply server/routes/rpaFramework.js:19]: Good point!
@kej-jay please do not use rhetorical questions as they are sometimes harder to understand.

**Review by lukashueller — APPROVED** [review]
Looks like a very careful job to me! I also like the wiki very much. Let's hope we get on well with the changeover.

**Comment by WolfgangDaniel** [comment]
> 
> 
> I am not a big fan of those dummy files. Please check, if there is any way to not have them in the repo, as they are not useful and might confuse people.
> 
> As my understanding of the client/api/ folder is that it shall include all API related functions, why are those calls from the front- to the backend not in there, but instead is a dummy file? Please let me know if I understood the purpose of that folder wrong or not.
> 
> Other than that only minor things that came up for me

File: client/src/components/content/PropertiesPanel/PropertiesPanelBuilder/PropertiesPanelBuilder.jsx (hunk 1 above)
Comment (WolfgangDaniel) [reply client/src/components/content/PropertiesPanel/PropertiesPanelBuilder/PropertiesPanelBuilder.jsx:120]: This is done by my vs code formatting on save. I would argue it is not a problem.

File: server/package.json (hunk 2 above)
Comment (WolfgangDaniel) [reply server/package.json:5]: You are right. I changed it to just server.

File: server/routes/rpaFramework.js (hunk 3 above)
Comment (WolfgangDaniel) [reply server/routes/rpaFramework.js:19]: I don't really get what specifically you are referring to with "styling"

File: server/server.js (hunk 4 above)
Comment (WolfgangDaniel) [reply server/server.js:33]: I deleted it :)

File: server/controllers/rpaFrameworkCommandsController.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
+const rpaFrameworkCommands = require('../models/rpaFrameworkCommands');
```
Comment (SanJSp) [inline server/controllers/rpaFrameworkCommandsController.js:1]: This is not used

**Review by SanJSp — COMMENTED** [review]
linter said that this is not used. Can you check this?

File: server/routes/rpaFramework.js (hunk 3 above)
Comment (WolfgangDaniel) [reply server/routes/rpaFramework.js:19]: Okay Lukas just told me what you meant with "styling". This styling is done automatically by my vs-code formatting on save. I use "Prettier - Code formatter" which is the must commonly used formatter on vs code.

File: server/controllers/rpaFrameworkCommandsController.js (hunk 5 above)
Comment (WolfgangDaniel) [reply server/controllers/rpaFrameworkCommandsController.js:1]: I put it in for demonstrating purposes but I can remove it now :)

**Review by kej-jay — APPROVED** [review]
